onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -324,7 +332,14 @@ - (Status)predict:(const std::unordered_map<std::string, OnnxTensorData>&)inputs
                                  ") do not match");
         }
 
-        const void* model_output_buffer = data.dataPointer;
+        ORT_RETURN_IF_NOT(IsArrayContiguous(data),
+                          "Non-contiguous output MLMultiArray is not currently supported");
+        __block const void* model_output_buffer = nil;
+        __block unsigned long coreml_buffer_size = 0;
+        [data getBytesWithHandler:^(const void* bytes, NSInteger size) {
+          model_output_buffer = bytes;
+          coreml_buffer_size = size;
```

Comment:
Hi! I believe you should use the `bytes` pointer while inside the handler and not pass it outside like this.

Reply:
Hi thanks for the feedback! I've been using it like this in the past as the docs don't specify any arc management nor lifetime for this pointer, so I assumed there were none. But I will fix it as you suggest as it seems safer in both cases!